pkg/kubelet/resources.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// defaultPodLimitsForDownwardApi copies the input pod, and optional container, and applies default resource limits.
+// it returns a copy of the input pod, and a copy of the input container (if specified) with default limits applied.
+// if a container has no limit specified, it will default the limit to the node capacity.
+// TODO: if/when we have pod level resources, we need to update this function to use those limits instead of node capacity.
+func (kl *Kubelet) defaultPodLimitsForDownwardApi(pod *api.Pod, container *api.Container) (*api.Pod, *api.Container, error) {
```

Comment:
the signature was done this way because it made it easier to plug-in the call structure here:
https://github.com/kubernetes/kubernetes/pull/27303/files#diff-bf28da68f62a8df6e99e447c4351122dR1563

In that location, I have no easy way of knowing what container in the pod is actually being referenced.

I think we can factor the calling area differently in the future.